rasa/core/channels/hangouts.py

```diff
@@ -0,0 +1,288 @@
+            new_messages = {"text": new_text}
+
+        elif msg_new == msg_state == "cards":
+            # two cards are combined into one
+            new_messages = self._combine_cards(self.messages, message)
+
+        elif msg_state == "cards" and msg_new == "text":
+            # if any message is card, turn text message into TextParagraph card and combine cards
+            text_card = self._text_card(message)
+            new_messages = self._combine_cards(self.messages, text_card)
+
+        elif msg_state == "text" and msg_new == "cards":
+            text_card = self._text_card(self.messages)
+            new_messages = self._combine_cards(text_card, message)
+
+        elif msg_new == "text":
+            new_messages = {"text": message.get("text")}
+        else:
+            new_messages = message
+
+        self.messages = new_messages
+
+    async def send_text_message(
+        self, recipient_id: Text, text: Text, **kwargs: Any
+    ) -> None:
+
+        await self._persist_message({"text": text})
+
+    async def send_image_url(self, recipient_id: Text, image: Text, **kwargs) -> None:
+
+        await self._persist_message(self._image_card(image))
+
+    async def send_text_with_buttons(
+        self, recipient_id: Text, text: Text, buttons: List, **kwargs
+    ) -> None:
+
+        await self._persist_message(self._text_button_card(text, buttons))
+
+    async def send_attachment(self, recipient_id: Text, attachment: Text):
+
+        await self.send_text_message(recipient_id, attachment)
+
+    async def send_elements(
+        self, recipient_id: Text, elements: Iterable[Dict[Text, Any]], **kwargs: Any
+    ) -> None:
+        raise NotImplementedError
+
+    async def send_custom_json(
+        self, recipient_id: Text, json_message: Dict, **kwargs
+    ) -> None:
+        """Custom json payload is simply forwarded to Google Hangouts without
+        any modifications. Use this for more complex cards, which can be created
+        in actions.py."""
+        await self._persist_message(json_message)
+
+
+# Google Hangouts input channel
+class HangoutsInput(InputChannel):
+    """
+    Channel that uses Google Hangouts Chat API to communicate.
+    """
+
+    @classmethod
+    def from_credentials(cls, credentials: Optional[Dict[Text, Any]]) -> InputChannel:
+        credentials = credentials or {}
+        return cls(
+            credentials.get("user_added_intent"),
+            credentials.get("room_added_intent"),
+            credentials.get("removed_intent"),
+        )
+
+    def __init__(
+        self,
+        user_added_intent: Optional[Text] = None,
+        room_added_intent: Optional[Text] = None,
+        removed_intent: Optional[Text] = None,
+    ) -> None:
+
+        self.user_added_intent = user_added_intent
+        self.room_added_intent = room_added_intent
+        self.removed_intent = removed_intent
+
+    @classmethod
+    def name(cls) -> Text:
+        return "hangouts"
```

Comment:
this name appears twice, maybe we can extract it to constant?

Reply:
Here I orientated myself on the other channels like facebook.py, where both InputChannel and OuputChanel need a name() classmethod and these also return the same value.

Reply:
you are right! I just meant that `hangouts` itself could be extracted to a const, and name() will return this constant.

Reply:
Done that

Reply:
that's exactly it, thanks!